examples/plugin-cli/plugins/mains/transit/go.mod

```diff
@@ -2,6 +2,10 @@ module github.com/hashicorp/go-kms-wrapping/examples/cli/plugins/mains/transit
 
 go 1.20
 
+replace github.com/hashicorp/go-kms-wrapping/plugin/v2 => ../../../../../plugin
```

Comment:
Why are using `replace` in the Go modules instead of importing the new repo in the OpenBao org?

Reply:
This is an in-tree import. Note that I've not renamed the entire module (from `hashicorp/go-kms-wrapping` to `openbao/go-kms-wrapping`), only fixed the dependency on `openbao/openbao/{api,sdk}`. 

This lets us avoid pushing a tag to this repository and then updating these dependent sub `go.mod` files to point to the new version when a new release is made. Since this is within the same repository space, pushing tags to make this work seems excessive and is used elsewhere (e.g., in the [main repository](https://github.com/openbao/openbao/blob/8dead566a8ab017c97a6417df33d326167eb5f57/go.mod#L15-L23)).

Reply:
IMO, I'd avoid doing the module rename now, as I'd prefer to see this imported into the main repository again. Upstream can support `go-kms-wrapping` as an externally-dependable module better than we can, and it remains MPL upstream. This way, we can focus on improving it in the context of OpenBao and not have to worry about breaking compatibility, IMHO.

Reply:
Ah, I had forgotten that sdk depended on it:

https://github.com/openbao/openbao/blob/8dead566a8ab017c97a6417df33d326167eb5f57/sdk/framework/backend.go#L21 

So we have to leave this as a separate repository for the time being.